### PR TITLE
Migrated to jupyter book v2

### PR DESCRIPTION
**Pull request by Jegp**
The recent failures to build the docs happened because Jupyter Book got bumped. This PR 
- Moves all config to `myst.yml`
- REMOVES the auto generated API - not yet supported in Jupyter Book 2
- Removes outdated roadmap information
- Adds the NIRData project, led by @benkroehs  :)

### CODE REVIEW

File: docs/source/api_nirtorch.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+This page lists functions and classes exposed by the `nirtorch` package for converting between PyTorch models and NIR.
+
+## Main Functions
+
+- **Graph extraction**: Convert PyTorch `nn.Module` to NIR graph
+- **Graph loading**: Convert NIR graph to PyTorch `nn.Module`
+
+For detailed usage examples and API documentation, see:
+- [PyTorch developer guide](dev_pytorch.ipynb)
+- [nirtorch source code](https://github.com/neuromorphs/nir/tree/main/nirtorch)
```
Comment (benkroehs) [inline docs/source/api_nirtorch.md]: This should be the NIRTorch repo, not NIR

https://github.com/neuromorphs/NIRTorch/tree/main/nirtorch

File: docs/source/api_nirtorch.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+## Key Modules
+
+- **Tracing**: Extract computational graphs from PyTorch models - see [tracing guide](nirtorch/tracing.ipynb)
```
Comment (benkroehs) [inline docs/source/api_nirtorch.md]: Better use the references with "#" here, like '#nirtorch_tracing'

File: docs/source/dev_pytorch.ipynb [hunk 3]
```diff
@@ -146,11 +147,11 @@
     "## Importing NIR models into PyTorch\n",
     "```{admonition} See also\n",
     ":class: seealso\n",
-    "Read more about importing NIR models into PyTorch on the page about [To PyTorch: Interpreting NIR](#nirtorch/interpreting).\n",
+    "Read more about importing NIR models into PyTorch on the page about [To PyTorch: Interpreting NIR](#nirtorch_interpreting).\n",
     "```\n",
     "\n",
-    "Assuming you have a NIR graph in the Python object `nir_graph` (see [Usage](#usage)), we need to inform `nirtorch` how to map NIR nodes into your simulator.\n",
-    "That is, for each node, we need a function (`nir.NIRNode -> torch.nn.Module`) which is given by a dictioary of type `Dictionary[nir.NIRNode, Callable[[nir.NIRNode], torch.nn.Module]]` (read about why in [To PyTorch: Interpreting NIR](#nirtorch/interpreting)).\n",
+    "Assuming you have a NIR graph in the Python object `nir_graph` (see [Usage](./usage.md)), we need to inform `nirtorch` how to map NIR nodes into your simulator.\n",
```
Comment (benkroehs) [inline docs/source/dev_pytorch.ipynb]: I think we prefer '#usage' over './usage.md'

**Review by benkroehs — COMMENTED** [review]
Thank for for updating the docs and fixing the deployment. The new documentation looks very nice. I have some minor comments but in general this looks fine.

File: docs/source/primitives.md [hunk 4]
```diff
@@ -97,7 +98,7 @@ NIR graphs should be self-contained and unambiguous, such that the graph itself
 
 ```{admonition} See also
 :class: info
-See [the usage page](usage) for more information on how to use NIR in practice.
+See [the usage page](./usage.md) for more information on how to use NIR in practice.
```
Comment (benkroehs) [inline docs/source/primitives.md]: better use #usage here

File: docs/source/roadmap.md [hunk 5]
```diff
@@ -10,53 +10,31 @@ We are dedicated to interacting and paying back to the community and we are cont
...
 **Owner**: [Bernhard Vogginger](https://github.com/bvogginger) </br>
 **Description**: The current implementation for SpiNNaker2 is static in the sense that we cannot stream input to the chip. We are working to support streaming inputs and, at the same time, speedup SNN processing. More information can be found at the [SpiNNaker 2 project milestone 5](https://gitlab.com/spinnaker2/py-spinnaker2/-/milestones/5).
 
-### Extending Lava & Loihi support
-**Duration**: October 2024 </br>
-**Owner**: [Steven Abreu](https://github.com/stevenabreu7/) </br>
-**Description**: Currently, Lava and Loihi support the bare minimum primitives for the paper. We are working to expand the support for neuron models and run additional applications.
+### Adding Nengo support
+**Owner**: [P. Michael Furlong](https://furlong.gitlab.io/) </br>
+**Description**: We are working to add Nengo support to NIR. This will allow users to export Nengo networks to NIR and for NIR users to run graphs via Nengo-supported hardware, such as Loihi and CPU. More information can be found at the [Nengo GitHub repository](https://furlong.gitlab.io/)
```
Comment (benkroehs) [inline docs/source/roadmap.md]: https://github.com/nengo/nengo

File: docs/source/api/index.md
[the remark this answers is not in the document]
Comment (Jegp) [reply docs/source/api/index.md]: Unfortunately, yes. The API generation is broken, so it's been replaced with static pages for now.

File: docs/source/api_nirtorch.md (hunk 1 above)
Comment (Jegp) [reply docs/source/api_nirtorch.md]: Great catch!

File: docs/source/dev_pytorch.ipynb (hunk 3 above)
Comment (Jegp) [reply docs/source/dev_pytorch.ipynb]: Yes! There was a problem using `#usage` because there was a headline in the README.md titled `Usage`, which was parsed as a label :face_exhaling: I fixed it by using another label. Nice catch.
